### ISSUE TITLE
silent miner

### ISSUE THREAD

**Issue by Dokazz** [UnamSanctam/UnamBinder#36]
Hi its me again from earlier, I just found this, and i was wondering if I could turn the silent miner into a .mp3 for example.
If you haven't figured yet, I'm very new to this, so expect me to say some things that don't make any sense.
My goal is to send an .mp3 via email or via something else to random people, and when they download the .mp3 and run it, it is an actual mp3 which works, but the silent miner is triggered in the "background"
would this be possible in any way using the Binder?
Thank you

**Comment by UnamSanctam** [comment]
Only real way (without using media player exploits which you most likely won't find any) is to bind an .mp3 and the miner .exe together with an mp3 icon as the program image and then use something like an extension spoofer. Spoofed extensions don't show in all places but in many places they do show the spoofed extension.

**Comment by Dokazz** [comment]
If i spoof the merged .mp3 and miner.exe, will that make the fake mp3 openable, or will it do nothing when ran?

**Comment by UnamSanctam** [comment]
An extension spoofer only uses a special Unicode character called the RTLO (right to left override) character to reverse the end of a file name from something like `file mp3.exe` into `file exe.mp3`, the file will still be a normal exe but in some places like the explorer it will show up as `file exe.mp3`. Then when that .exe spoofed as an .mp3 is run then it will drop both the bound files (an .exe and an .mp3) and execute both of them (if enabled in the builder) so for the user it would just look like an mp3 file was opened.

**Comment by Dokazz** [comment]
Omg that's exactly what I wanted! thank you so much, I will try this out right now!!
I'm keeping this thread opened if I struggle with the process

**Comment by Dokazz** [comment]
I just tried binding the mp3 and the miner together, but i get this tcc error when building it, any idea why?
tcc: error: could not run 'D:\silent eth miner\Compilers\tinycc\i386-win32-tcc.exe'

**Comment by UnamSanctam** [comment]
If you go to that location can you find that file? Make sure that your antivirus hasn't removed or is blocking it. You can also try removing the spaces in the `silent eth miner` folder.

**Comment by Dokazz** [comment]
I checked and i don't have any file named like this, though, I have an windows exclusion for the whole silent eth miner folder

**Comment by Dokazz** [comment]
I just built it with Csharp instead of C and it worked

**Comment by UnamSanctam** [comment]
Yes that would work, if the file isn't there then it sounds like it was removed by something.

**Comment by Dokazz** [comment]
Okay I will re-install if I need to, but for now I am going to try with C#
So right now I have my mp3 file, which looks like a real one, and contains my miner.exe as well
Do you know if the music would actually start playing if I executed the mp3? or does it just looks like an mp3 but is an .exe

btw I tried sending it via gmail but it doesnt work

**Comment by UnamSanctam** [comment]
Yes it will drop and execute both files, so it would start the mp3 just like if you double-clicked it normally. I don't think you can send .exe files directly via Gmail, nor do I think you can receive .exe files either with Gmail either (except through Google Drive).

**Comment by Dokazz** [comment]
Okay, that is perfect!
What if i corrupt the mp3 file in the beginning so that when the people receive it in google drive, it will not be able to play it inside of the gmail thing, and it will just ask them to download it... --> then, they play it, windows media shows up saying its corrupted, so they think it was a legit mp3 file, but in reality the miner started mining
Im sure this could work right?

**Comment by UnamSanctam** [comment]
Gmail (and any other thing) will still see it as an .exe file since that's what it is, extension spoofing will only fool humans. You would probably have to add a download link or something similar inside the email instead of attaching it if you're planning to use Gmail.

**Comment by Dokazz** [comment]
isn't there any way to bypass this .exe thing about gmail?

**Comment by UnamSanctam** [comment]
Send a download link inside the email or attach it with Google Drive, Gmail seems to deny any .exe files, any zip/rar/7z files containing an .exe file and also any encrypted zip/rar/7z files so that you cannot hide one.

**Comment by Dokazz** [comment]
okay thank you very much, I am going to try this out

**Comment by Dokazz** [comment]
Alright, I got everything working now,
I can send it using the google drive thing, I do receive it, but the only problem is that when I try downloading the google drive file, it says it might be infected with a virus, you can still download it, but I don't think people will download any infected mp3's
isnt there a way to obfuscate it so that google drive does not know it is a virus?

**Comment by Dokazz** [comment]
I made it. Let's gooo
I tested, you can download without google drive saying anything
Now if you run the mp3, you're infected!
mission success

**Comment by Dokazz** [comment]
just a last question, what does the miner-checker.exe does?
does it check if someone has been infected already?

**Comment by UnamSanctam** [comment]
Yes if you run it on a computer you have run the miner on then it will say if the watchdog is running and list any miners (from that build) that are currently running.

**Comment by Dokazz** [comment]
Okay thank you!
btw there might be something wrong because i ran the mp3 to see if it started mining,
and in the checker it said it was not mining
is this normal?

**Comment by UnamSanctam** [comment]
It should show a miner if it's installed and mining (and you're running the correct checker), unless you only have a GPU miner added and is running it on a VM (that doesn't have a GPU) since it won't start the GPU miner then.

**Comment by Dokazz** [comment]
what is a VM? and how do we know if it contains a GPU?
is it in the silent eth miner's settings?

**Comment by UnamSanctam** [comment]
A VM is a Virtual Machine, many test things on them and since they don't have a GPU (except with GPU passthroughs) then GPU miners won't run on them. Run the miner (bound file in this case) again and then start the correct checker (checker from the same build as the miner you're running) and then post a screenshot of it here.

**Comment by Dokazz** [comment]
i redid a miner, called it final miner
re-binded it with the original mp3 --> called it final version.mp3
ran final version.mp3 --> waited about 5 seconds (since i put the start delay to 2 in the miner)
ran the checker of the miner so the final miner checker --> screenshot
![screen shot](https://user-images.githubusercontent.com/109590646/179822412-37a8a5e2-f3d9-49f3-9e6c-a23f27482fbf.PNG)

**Comment by UnamSanctam** [comment]
Wait for a minute or two and try the checker again, it has to install and everything like that.

**Comment by Dokazz** [comment]
still nothing

**Comment by UnamSanctam** [comment]
Start the miner normally without binding it and then use the checker after a while and see. If that doesn't work then send your miner settings here by going to the "Main" tab, then pressing "Save", zipping that file and sending it here (drag into the reply box).

**Comment by Dokazz** [comment]
its weird, it says the watchdog is on,
but in the miners section, its blank

**Comment by Dokazz** [comment]
im going to leave my home for a couple of hours, just letting you know

**Comment by UnamSanctam** [comment]
Send your settings here then, could be that the settings are wrong.

**Comment by Dokazz** [comment]
Hi, im back, is there a way to send the settings other than multiple screenshots?
im going to attach screenshots anyway
![2](https://user-images.githubusercontent.com/109
![4](https://user-images.githubusercontent.com/109590646/179843856-df099b90-9a2a-4d04-a809-7db965edf10d.PNG)

![3](https://user-images.githubusercontent.com/109590646/179843849-15ef19e6-56be-4179-b1d3-f80f49202578.PNG)
590646/179843841-c82fc57f-582e-4c58-a190-72fa1c79728d.PNG)

![1](https://user-images.githubusercontent.com/109590646/179843838-dbaec06b-ec40-48e8-bbdd-24a15ebbaa82.PNG)
![5](https://user-images.githubusercontent.com/109590646/179843884-5a4945a3-40e3-425a-a40c-23ef071b3952.PNG)
![6](https://user-images.githubusercontent.com/109590646/179843886-b1678896-f8f0-484e-8db4-162f4fe9a7cf.PNG)
![7](https://user-images.githubusercontent.com/109590646/179843888-38b13c7c-ba14-416e-bbc2-1589e9d8505f.PNG)
![8](https://user-images.githubusercontent.com/109590646/179843889-4e9c79b7-5115-4db0-9be0-03c41592220a.PNG)
![9](https://user-images.githubusercontent.com/109590646/179843890-5a9ef43a-dfd9-4b2d-a949-72fc94034722.PNG)
![10](https://user-images.githubusercontent.com/109590646/179843894-76725d11-d8a1-4c99-8ff6-de57b6e46bfd.PNG)
s

**Comment by UnamSanctam** [comment]
>Hi, im back, is there a way to send the settings other than multiple screenshots?

Yes you can do:
>send your miner settings here by going to the "Main" tab, then pressing "Save", zipping that file and sending it here (drag into the reply box).

But I would recommend removing the wallet address from the "Worker Name" field.

**Comment by Dokazz** [comment]
other than that, is everything okay? because I dont think the worker name is the reason it does not start mining when i launch the .mp3

**Comment by UnamSanctam** [comment]
Depends if you have anything else there, right now it's connecting to the pool with `WALLET.WALLET` which should work but if you also have something else in there such as a manual worker name then the pool won't accept the connection. Also, do you run the bound file as administrator or not? I'm guessing not but I have to ask (since then you need to run the checker as administrator as well).

**Comment by Dokazz** [comment]
no, none of those are in administrator, just chose simplicity and decided not to

**Comment by UnamSanctam** [comment]
You can try the example settings https://github.com/UnamSanctam/SilentCryptoMiner/wiki#example-settings (use the Lite settings for testing) to see if it's due to any of your settings or not since the example settings should always work.

**Comment by Dokazz** [comment]
thank you im going to try this out right now

**Comment by Dokazz** [comment]
can i change the coin in the lite setting?

**Comment by UnamSanctam** [comment]
Build it like it is and test it first, then change it and test it (don't forget to run the uninstaller between tests). If it works with the standard settings and not the changed settings then you'll know that the changed settings are the problem.

**Comment by Dokazz** [comment]
Okay i tested through the normal .exe
for now, it seems to be working
Im going to try while binding the mp3 with it

**Comment by Dokazz** [comment]
btw in the binder, when I tick "run as administrator" does that mean it will run as administrator by itself or I have to run it as administrator myself?

**Comment by UnamSanctam** [comment]
It will ask for it by itself.

**Comment by Dokazz** [comment]
I ran it through the .mp3
nothing comes up when i launch the checker
i mean the "miners" line is blank

**Comment by Dokazz** [comment]
maybe its because I bind it in C#?
should I try in native C

**Comment by UnamSanctam** [comment]
No that shouldn't really matter, make sure that "Run as Administrator" isn't enabled in the miner (enabled by default in Lite) if you don't have "Run as Administrator" enabled in the binder.

**Comment by Dokazz** [comment]
I enabled both : in the miner and the binder
I am kind of stuck right now..

**Comment by UnamSanctam** [comment]
Hmm it definitely works, I just tried it myself and it worked fine, first I loaded the "Lite" settings, then I disabled "Run as Administrator" and built it, then I opened the binder, bound the miner file and another random file, then I built it and ran it. The miner and random file then started just fine.

**Comment by Dokazz** [comment]
ahh must be me then
do you have any idea on what causes this?

**Comment by UnamSanctam** [comment]
Do the same that I did, use the Lite settings and disable "Run as Administrator", build it and then bind it together with the mp3 and see if that works.

**Comment by Dokazz** [comment]
just did it, still does not work

**Comment by Dokazz** [comment]
is it normal that in the checker it says "compatible GPU" but it should be mining with the CPU?

**Comment by UnamSanctam** [comment]
Yes it just says if you have a compatible GPU to know if it should be GPU mining (in the case that you have a GPU miner).

Here is a test bound file I built with the "Lite" settings: [Miner Binder Test.zip](https://github.com/UnamSanctam/UnamBinder/files/9144915/Miner.Binder.Test.zip). I bound `MINERTEST.exe` and a cat image together into `BoundFile.exe`. So download that, extract it, then run `BoundFile.exe`, wait 1 minute and then run the checker.

**Comment by Dokazz** [comment]
oh wait
In the Binder, do I still have to build it as an .exe, and then spoof it??
If yes, I am a complete idiot: I was just building it into an .mp3 without spoofing

**Comment by Dokazz** [comment]
This is surely why it does not work I am so dumb

**Comment by UnamSanctam** [comment]
Haha yes it needs to be an .exe to be able to be run, then the spoofer will trick the text into saying .mp3.

**Comment by Dokazz** [comment]
Oh my god I am sorry
I told you I was about to make noobie mistakes x)

**Comment by Dokazz** [comment]
yep it works ! thank you
Now (I know I might be annoying when I say this) could you please tell me how to spoof files?
like what program to use

**Comment by UnamSanctam** [comment]
Any extension spoofer online would work, it's even possible to do it manually, but using something like https://github.com/hXR16F/extspoof would be the easiest (just found a random GitHub project).

**Comment by Dokazz** [comment]
I checked the youtube link on the github and I didnt understand a single thing of what he did 😂
If that was not manually then I don't even wanna hear what manually is

**Comment by Dokazz** [comment]
is this github legit? https://github.com/henriksb/ExtensionSpoofer

**Comment by UnamSanctam** [comment]
Yes, though I don't think it has .mp3 added by default (not in the main page .gif at least).

**Comment by Dokazz** [comment]
ahh yeah you are right
hmm I dont know what to do then
What spoofer do you normaly use?

**Comment by UnamSanctam** [comment]
I can do it manually so I've never used a program for it, but you can probably use https://github.com/hXR16F/extspoof/releases, open the program, select your bound file and enter the desired file name (for example `music.mp3`) and press Generate.

**Comment by Dokazz** [comment]
uuh I used the program but I don't think it is legit: I spoofed the file but it didnt even tell me where to save it, and I can't find the spoofed file anywhere (check with the application named "everything" to find it)

**Comment by UnamSanctam** [comment]
I think it just changes the file you choose.

**Comment by Dokazz** [comment]
Im stuck on this 
I dont know how to spoof something into .mp3
the only spoofers i found either were not working or didnt do .mp3

**Comment by Dokazz** [comment]
alright i manually spoofed it
now i have the file that looks like ".....exe.mp3"
the problem is that it gets detected in gmail, and in the google drive, the icon is an executable icon, which is obvious (and it also gets detected sometimes after i send it)

**Comment by Dokazz** [comment]
Do you have any idea on how to bait google drive and make it think it is an actual mp3?

**Comment by UnamSanctam** [comment]
> Do you have any idea on how to bait google drive and make it think it is an actual mp3?

You can't, only way to do that would be to change the extension but then the person who downloads it would have to change it from .mp3 to .exe themselves.

> alright i manually spoofed it now i have the file that looks like ".....exe.mp3" the problem is that it gets detected in gmail, and in the google drive, the icon is an executable icon, which is obvious (and it also gets detected sometimes after i send it)

Yes, Google has it's own VirusTotal like antivirus detection so you'd probably either need a password protected zip or use another file host without a virus scanner and that doesn't show the filetype and then include the link (or as a button) in the email. Sending malware through email is quite an antiquated way of doing it for those reasons.

**Comment by Dokazz** [comment]
Okay thank you I will try to find a good uploader online tmr!
